src/compas/datastructures/assembly/part.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+
+from collections import deque
+from functools import reduce
+
+from compas.geometry import Shape
+from compas.geometry import Frame
+from compas.geometry import multiply_matrices
+from compas.geometry import boolean_union_mesh_mesh
+from compas.geometry import boolean_difference_mesh_mesh
+from compas.geometry import boolean_intersection_mesh_mesh
+
+from ..datastructure import Datastructure
+from ..mesh import Mesh
+
+from .exceptions import FrameError
+from .exceptions import FeatureError
+
+
+class Part(Datastructure):
+    """A data structure for representing assembly parts."""
+
+    operations = {
+        'union': boolean_union_mesh_mesh,
+        'difference': boolean_difference_mesh_mesh,
+        'intersection': boolean_intersection_mesh_mesh
+    }
+
+    @property
+    def DATASCHEMA(self):
+        import schema
+        return schema.Schema({
+            "attributes": dict,
+            "key": int,
+            "frame": Frame,
+            "transformations": list,
+            "geometry": Shape,
+            "features": list
+        })
+
+    @property
+    def JSONSCHEMANAME(self):
+        return 'assembly'
+
+    def __init__(self, name, **kwargs):
+        super(Part, self).__init__()
+        self._key = None
+        self._frame = None
+        self._transformations = deque()
+        self._geometry = None
+        self._features = []
+        self._shape = None
+
+    def __str__(self):
+        tpl = "<Part with base geometry {} and features {}>"
+        return tpl.format(self.geometry, self.features)
+
+    @property
+    def data(self):
+        """dict : A data dict representing the part attributes, internal data structure, and geometries for serialization.
+        """
+        data = {
+            'attributes': self.attributes,
+            'mesh': self.mesh.data,
+            'graph': self.graph.data
+        }
+        return data
+
+    @data.setter
+    def data(self, data):
+        self.attributes.update(data['attributes'] or {})
+        self.mesh.data = data['mesh']
+        self.graph.data = data['graph']
```

Comment:
these two methods look like left overs of a previous implementation, right? (mesh and graph are no longer here)

Reply:
indeed ... updated!